great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -37,14 +37,12 @@
     Datasource,
     TestConnectionError,
 )
+from great_expectations.util import NotImported
 
-SQLALCHEMY_IMPORTED = False
 try:
     import sqlalchemy
-
-    SQLALCHEMY_IMPORTED = True
 except ImportError:
-    pass
+    sqlalchemy = NotImported("sqlalchemy not found, please install.")
```

Comment:
The import strategy looks similar to what we do now except instead of setting the module to `None` you set it to `NotImported` which will result in better error messages. I was hoping one could avoid the try/except block but that proved hard because of how mypy handles the interplay of dynamic types that might not be types when not type-checking but will be types when type-checking.